resources/keycloak/custom-user-provider/src/main/java/com/cvmanager/auth/provider/user/UserAdapter.java

```diff
@@ -0,0 +1,249 @@
+    protected UserObject entity;
+    protected String keycloakId;
+    final CustomUserStorageProvider provider;
+    final RealmModel realm;
+
+    public UserAdapter(KeycloakSession session, RealmModel realm, ComponentModel model, UserObject entity) {
+        super(session, realm, model);
+        this.entity = entity;
+        keycloakId = StorageId.keycloakId(model, entity.getId());
+        provider = new CustomUserStorageProvider(session, model);
+        this.realm = realm;
+    }
+
+    public Integer getSuperUser() {
+        return entity.getSuperUser();
+    }
+
+    public void setSuperUser(Integer superUser) {
+        entity.setSuperUser(superUser);
+    }
+
+    public List<OrganizationObject> getOrganizations() {
+        return entity.getOrganizations();
+    }
+
+    public void setOrganizations(List<OrganizationObject> organizations) {
+        entity.setOrganizations(organizations);
+    }
+
+    @Override
+    public String getUsername() {
+        return entity.getEmail();
+    }
+
+    @Override
+    public void setUsername(String username) {
+        entity.setEmail(username);
+    }
+
+    @Override
+    public String getEmail() {
+        return entity.getEmail();
+    }
+
+    @Override
+    public void setEmail(String email) {
+        entity.setEmail(email);
+    }
+
+    @Override
+    public String getFirstName() {
+        return entity.getFirstName();
+    }
+
+    @Override
+    public void setFirstName(String firstName) {
+        entity.setFirstName(firstName);
+    }
+
+    @Override
+    public String getLastName() {
+        return entity.getLastName();
+    }
+
+    @Override
+    public void setLastName(String lastName) {
+        entity.setLastName(lastName);
+    }
+
+    @Override
+    public Long getCreatedTimestamp() {
+        return entity.getCreatedTimestamp();
+    }
+
+    @Override
+    public void setCreatedTimestamp(Long timestamp) {
+        entity.setCreatedTimestamp(timestamp);
+    }
+
+    @Override
+    public String getId() {
+        return keycloakId;
+    }
+
+    @Override
+    public void setSingleAttribute(String name, String value) {
+        log.debug("Setting attribute: " + name + " to " + value);
+        switch (name) {
+            case UserModel.USERNAME:
+                entity.setEmail(value);
+                break;
+            case UserModel.EMAIL:
+                entity.setEmail(value);
+                break;
+            case UserModel.FIRST_NAME:
+                entity.setFirstName(value);
+                break;
+            case UserModel.LAST_NAME:
+                entity.setLastName(value);
+                break;
+            case "created_timestamp":
+                entity.setCreatedTimestamp(value == null ? null : Long.valueOf(value));
+                break;
+            case "super_user":
+                entity.setSuperUser(value == null ? null : Integer.valueOf(value));
+                break;
+            case "organizations":
```

Comment:
Suggestion (non-blocking): Should we consider making these strings into constants to reduce duplication (and usage of magic strings)?

Reply:
updated